### ISSUE TITLE
GitHub Actions backend: Builds aren't parallelized?

### ISSUE THREAD

**Issue by RyanGlScott** [haskell-CI/haskell-ci#421]
When building dependencies in `haskell-ci`–generated GitHub Actions configuration, the installation completely lacks any parallelism. ([Example](https://github.com/goldfirere/singletons/pull/479/checks?check_run_id=1557542033#step:16:66).) I wonder if the backend needs to use `-j2` somewhere, like what the Travis backend uses?

**Comment by RyanGlScott** [comment]
The Bash backend, judging from the source code, seems to suffer from a similar problem. Curiously, `-j2` is used when installing `doctest`:

https://github.com/haskell-CI/haskell-ci/blob/e991d01d51aeeb551d2d15c3833e7988b8eecdb7/src/HaskellCI/Bash.hs#L44

But nowhere else.

**Comment by phadej** [comment]
Yes. With `bash` it's tricky, as it depends on users machine/current use, which `-j` is reasonable. For GitHub Actions we should use whatever cores GHA provides. Please find the environment info, and add/change according `-j`.

The actual build is better to keep `-j1` though. Why? Because interleaved output is hard to parse. This can be configurable, but default should be no parallelism.

**Comment by RyanGlScott** [comment]
> For GitHub Actions we should use whatever cores GHA provides. Please find the environment info, and add/change according `-j`.

According to [this](https://docs.github.com/en/free-pro-team@latest/actions/reference/specifications-for-github-hosted-runners#supported-runners-and-hardware-resources), GitHub Actions virtual machines use 2-core CPUs, so `-j2` seems reasonable.

> The actual build is better to keep `-j1` though. Why? Because interleaved output is hard to parse. This can be configurable, but default should be no parallelism.

Absolutely. To be clear, when I say "parallelism", I'm exclusively referring to installation of dependencies, not building the main packages themselves.
